request.go

```diff
@@ -1,20 +1,42 @@
 package ping
 
 import (
+	"net"
+	"sync"
 	"time"
 )
 
-// A request is a currently running ICMP echo request waiting for an answer.
-type request struct {
+type request interface {
+	init()
+	close()
+	handleResponse(error, net.IP, *time.Time)
+}
+
+// A request is a currently running ICMP echo request waiting for multple answers.
+type multiRequest struct {
+	tStart    time.Time // when was the request packet sent?
+	responses chan Response
+	closed    bool
+	mtx       sync.RWMutex
+}
+
+// Response is a response to a multicast echo request
+type Response struct {
```

Comment:
A response to an echo request is commonly known as "Reply" → `type Reply`?

Reply:
Thanks, has been renamed.